misk-hibernate/src/main/kotlin/misk/hibernate/QueryTracingListener.kt

```diff
@@ -0,0 +1,110 @@
+package misk.hibernate
+
+import io.opentracing.Scope
+import io.opentracing.Tracer
+import misk.logging.getLogger
+import org.hibernate.event.spi.PostDeleteEvent
+import org.hibernate.event.spi.PostDeleteEventListener
+import org.hibernate.event.spi.PostInsertEvent
+import org.hibernate.event.spi.PostInsertEventListener
+import org.hibernate.event.spi.PostLoadEvent
+import org.hibernate.event.spi.PostLoadEventListener
+import org.hibernate.event.spi.PostUpdateEvent
+import org.hibernate.event.spi.PostUpdateEventListener
+import org.hibernate.event.spi.PreDeleteEvent
+import org.hibernate.event.spi.PreDeleteEventListener
+import org.hibernate.event.spi.PreInsertEvent
+import org.hibernate.event.spi.PreInsertEventListener
+import org.hibernate.event.spi.PreLoadEvent
+import org.hibernate.event.spi.PreLoadEventListener
+import org.hibernate.event.spi.PreUpdateEvent
+import org.hibernate.event.spi.PreUpdateEventListener
+import org.hibernate.persister.entity.EntityPersister
+import javax.inject.Singleton
+
+private val logger = getLogger<QueryTracingListener>()
+
+@Singleton
+internal class QueryTracingListener
+  : PreLoadEventListener,
+    PostLoadEventListener,
+    PreInsertEventListener,
+    PostInsertEventListener,
+    PreUpdateEventListener,
+    PostUpdateEventListener,
+    PreDeleteEventListener,
+    PostDeleteEventListener {
+
+  @com.google.inject.Inject(optional = true) var tracer: Tracer? = null
+
+  private val lastScopeInThread: ThreadLocal<Scope?> = ThreadLocal.withInitial { null }
+
+  override fun onPreInsert(event: PreInsertEvent?): Boolean {
+    startSpan(DB_INSERT)
+    return false
+  }
+
+  override fun onPostInsert(event: PostInsertEvent?) {
+    endLastSpan()
+  }
+
+  override fun onPreUpdate(event: PreUpdateEvent?): Boolean {
+    startSpan(DB_UPDATE)
+    return false
+  }
+
+  override fun onPostUpdate(event: PostUpdateEvent?) {
+    endLastSpan()
+  }
+
+  override fun onPreDelete(event: PreDeleteEvent?): Boolean {
+    startSpan(DB_DELETE)
+    return false
+  }
+
+  override fun onPostDelete(event: PostDeleteEvent?) {
+    endLastSpan()
+  }
+
+  override fun onPreLoad(event: PreLoadEvent?) {
+    startSpan(DB_SELECT)
+  }
+
+  override fun onPostLoad(event: PostLoadEvent?) {
+    endLastSpan()
+  }
+
+  override fun requiresPostCommitHanding(persister: EntityPersister?): Boolean {
+    return false
+  }
+
+  private fun startSpan(spanName: String) {
+    if (tracer == null) {
+      return
+    }
+
+    val lastScope = lastScopeInThread.get()
+    lastScope?.let {
+      logger.debug { "lastScope wasn't closed" }
```

Comment:
would we want at least a logger.info with some debugging information? seems like a bug we'd want to know is occurring/track down

Reply:
would just want to rate limit the logging.